### PR TITLE
Issue 17059: Initial drop of OIDC client JWE support

### PR DESCRIPTION
**Pull request by ayoho**
For issue #17059
For issue #17060
    
Adds partial support for:
- Verifying ID tokens in JWE format returned from an OP
- Verifying access tokens in JWE format in inbound propagation scenarios

### CODE REVIEW

**Comment by ayoho** [comment]
Personal build running: https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_1DTRYMOqEeu8guqBTZau2w

**Comment by skasund** [comment]
L2 message review completed

**Comment by ayoho** [comment]
#libby
#build
#spawn.fullfat.buckets=com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.jwtsso_fat,com.ibm.ws.security.mp.jwt.1.1_fat,com.ibm.ws.security.mp.jwt.1.2_fat,com.ibm.ws.security.oidc.client_fat,com.ibm.ws.security.oidc.client_fat.config,com.ibm.ws.security.oidc.client_fat.e2e,com.ibm.ws.security.oidc.client_fat.jaxrs,com.ibm.ws.security.oidc.client_fat.saml,com.ibm.ws.security.oidc.client_fat.spnego,com.ibm.ws.security.oidc.server_fat,com.ibm.ws.security.oidc.server_fat.app-passwords,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-passwords.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.DerbyDB.opaque,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.jwt,com.ibm.ws.security.oidc.server_fat.app-tokens.LocalStore.opaque,com.ibm.ws.security.oidc.server_fat.clientregistration,com.ibm.ws.security.oidc.server_fat.config,com.ibm.ws.security.oidc.server_fat.config.oauth,com.ibm.ws.security.oidc.server_fat.config.oidc,com.ibm.ws.security.oidc.server_fat.endpoint.authorize,com.ibm.ws.security.oidc.server_fat.endpoint.coveragemap,com.ibm.ws.security.oidc.server_fat.endpoint.discovery,com.ibm.ws.security.oidc.server_fat.endpoint.end_session,com.ibm.ws.security.oidc.server_fat.endpoint.introspect,com.ibm.ws.security.oidc.server_fat.endpoint.revoke,com.ibm.ws.security.oidc.server_fat.endpoint.token,com.ibm.ws.security.oidc.server_fat.endpoint.userinfo,com.ibm.ws.security.oidc.server_fat.feature,com.ibm.ws.security.oidc.server_fat.granttype.jwt,com.ibm.ws.security.oidc.server_fat.gui.tokenMgmt,com.ibm.ws.security.oidc.server_fat.jaxrs,com.ibm.ws.security.oidc.server_fat.jaxrs.config,com.ibm.ws.security.oidc.server_fat.oauth,com.ibm.ws.security.oidc.server_fat.oidc,com.ibm.ws.security.oidc.server_fat.saml,com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.social_fat.OpenShift,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.multiProvider

File: dev/com.ibm.ws.security.openidconnect.clients.common/src/com/ibm/ws/security/openidconnect/client/jose4j/util/Jose4jUtil.java [hunk 1]
```diff
@@ -196,103 +197,19 @@ boolean useAccessTokenAsIdToken(ConvergedClientConfig clientConfig) {
...
-    //            customProperties.put(ClientConstants.ISSUED_JWT_TOKEN, jwt);
-    //        }
-    //
-    //    }
-
     //Just parse without validation for now
-    protected static JwtContext parseJwtWithoutValidation(String jwtString) throws Exception {
+    protected static JwtContext parseJwtWithoutValidation(String jwtString, ConvergedClientConfig clientConfig) throws Exception {
+        String jwtStringToProcess = jwtString;
+        if (JweHelper.isJwe(jwtString)) {
```
Comment (teddyjtorres) [inline dev/com.ibm.ws.security.openidconnect.clients.common/src/com/ibm/ws/security/openidconnect/client/jose4j/util/Jose4jUtil.java:203]: We need to add a beta guard in this if statement.

File: dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/config/JweConsumerConfig.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*******************************************************************************
```
Comment (teddyjtorres) [inline dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/config/JweConsumerConfig.java:1]: It appears this interface is not used yet. The code could be simplified by removing this interface and adding the methods back to the JwtConsumerConfig interface and adding also getJweDecryptionKey() to it, making it the authority of all things related to JWT consumption, including JWE.

Comment (ayoho) [reply dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/config/JweConsumerConfig.java:1]: My follow up to this delivery that I already have ready to go does exactly that.

**Pull request merged**